File: backend/app/schemas/publishing.py
from typing import Optional

from pydantic import BaseModel


class PublishPostRequest(BaseModel):
    post_id: int
    social_account_id: int
    content_text: Optional[str] = None
    hashtags: Optional[str] = None


class PublishPostResponse(BaseModel):
    success: bool
    platform: str
    social_account_id: int
    post_id: int
    published_id: Optional[str] = None
    published_url: Optional[str] = None
    message: str

File: frontend/src/types/publishing.ts
export type PublishPostPayload = {
  post_id: number;
  social_account_id: number;
  content_text?: string;
  hashtags?: string;
};

export type PublishPostResponse = {
  success: boolean;
  platform: string;
  social_account_id: number;
  post_id: number;
  published_id?: string | null;
  published_url?: string | null;
  message: string;
};
